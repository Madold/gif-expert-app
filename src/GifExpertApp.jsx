import { useState } from "react";
import { AddCategory, GifGrid } from "./components";

export const GifExpertApp = () => {

    const [categories, setCategories] = useState([]);

    const addcategory = (categorie) => {
        if(isRepeated(categorie)) return

        setCategories([categorie, ...categories])
    }

    const isRepeated = (category) => categories.map(it => it.toLowerCase()).includes(category.toLowerCase())

    return (
        <>
            <h1>Gif ExpertApp</h1>
            <hr />
            <AddCategory onNewCategory={ addcategory }/>
            <ol>
                {
                    categories.map(category => <li key={category}><GifGrid category={category}/></li>)
                }
            </ol>
        </>
    )
}


