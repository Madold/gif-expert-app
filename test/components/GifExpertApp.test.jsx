import {fireEvent, render, screen} from '@testing-library/react'
import {GifExpertApp} from "../../src/GifExpertApp"



describe('Pruebas en GifExpertApp', () => {
    
    test('Debe hacer match con el snapshot', () => {
         const {container} =  render(<GifExpertApp />);
         expect(container).toMatchSnapshot();
    });
    
    test("Debe mostrar una categoría", () => {
        const category = "One Punch"
        render(<GifExpertApp />);
        const form = screen.getByRole("form");
        const input = screen.getByRole("textbox");
        fireEvent.input(input, { target: { value: category } });
        fireEvent.submit(form);
        const title = screen.getByText(category);
        screen.debug();
        expect(title).toBeTruthy();

    })

    test('Una categoría repetida no debe ser añadida', () => {
        const category = "One Punch"
        render(<GifExpertApp />);
        const form = screen.getByRole("form");
        const input = screen.getByRole("textbox");
        fireEvent.input(input, { target: { value: category } });
        fireEvent.submit(form);
        fireEvent.input(input, { target: { value: category } });
        fireEvent.submit(form);
        const title = screen.getAllByText(category);
        expect(title.length).toBe(1);
    });
    

});
