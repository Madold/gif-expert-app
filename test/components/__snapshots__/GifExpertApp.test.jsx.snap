// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en GifExpertApp Debe hacer match con el snapshot 1`] = `
<div>
  <h1>
    Gif ExpertApp
  </h1>
  <hr />
  <form
    aria-label="form"
  >
    <input
      placeholder="Buscar gif"
      type="text"
      value=""
    />
  </form>
  <ol />
</div>
`;
